pkg/renderer/sgml/html5/templates.go

```diff
@@ -0,0 +1,84 @@
+package html5
+
+import (
+	"github.com/bytesparadise/libasciidoc/pkg/renderer/sgml"
+	"io"
+)
+
+var templates = &sgml.Templates{
+	AdmonitionBlock:         admonitionBlockTmpl,
+	AdmonitionParagraph:     admonitionParagraphTmpl,
+	Article:                 articleTmpl,
+	ArticleHeader:           articleHeaderTmpl,
+	BlankLine:               blankLineTmpl,
+	BlockImage:              blockImageTmpl,
+	BoldText:                boldTextTmpl,
+	CalloutList:             calloutListTmpl,
+	DelimitedBlockParagraph: delimitedBlockParagraphTmpl,
+	DocumentDetails:         documentDetailsTmpl,
+	DocumentAuthorDetails:   documentAuthorDetailsTmpl,
+	ExternalCrossReference:  externalCrossReferenceTmpl,
+	ExampleBlock:            exampleBlockTmpl,
+	FencedBlock:             fencedBlockTmpl,
+	Footnote:                footnoteTmpl,
+	FootnoteRef:             footnoteRefTmpl,
+	FootnoteRefPlain:        footnoteRefPlainTmpl,
+	Footnotes:               footnotesTmpl,
+	InlineImage:             inlineImageTmpl,
+	InternalCrossReference:  internalCrossReferenceTmpl,
+	InvalidFootnote:         invalidFootnoteTmpl,
+	ItalicText:              italicTextTmpl,
+	LabeledList:             labeledListTmpl,
+	LabeledListHorizontal:   labeledListHorizontalTmpl,
+	LineBreak:               lineBreakTmpl,
+	Link:                    linkTmpl,
+	ListingBlock:            listingBlockTmpl,
+	LiteralBlock:            literalBlockTmpl,
+	ManpageHeader:           manpageHeaderTmpl,
+	ManpageNameParagraph:    manpageNameParagraphTmpl,
+	MonospaceText:           monospaceTextTmpl,
+	OrderedList:             orderedListTmpl,
+	PassthroughBlock:        pssThroughBlock,
+	Paragraph:               paragraphTmpl,
+	Preamble:                preambleTmpl,
+	QAndAList:               qAndAListTmpl,
+	QuoteBlock:              quoteBlockTmpl,
+	QuoteParagraph:          quoteParagraphTmpl,
+	SectionContent:          sectionContentTmpl,
+	SectionHeader:           sectionHeaderTmpl,
+	SectionOne:              sectionOneTmpl,
+	SidebarBlock:            sidebarBlockTmpl,
+	SourceBlock:             sourceBlockTmpl,
+	SourceBlockContent:      sourceBlockContentTmpl,
+	SourceParagraph:         sourceParagraphTmpl,
+	StringElement:           stringTmpl,
+	SubscriptText:           subscriptTextTmpl,
+	SuperscriptText:         superscriptTextTmpl,
+	Table:                   tableTmpl,
+	TocRoot:                 tocRootTmpl,
+	TocSection:              tocSectionTmpl,
+	UnorderedList:           unorderedListTmpl,
+	VerbatimLine:            verbatimLineTmpl,
+	VerseBlock:              verseBlockTmpl,
+	VerseBlockParagraph:     verseBlockParagraphTmpl,
+	VerseParagraph:          verseParagraphTmpl,
+}
+
+var defaultRenderer sgml.Renderer
```

Comment:
I'm not sure we really need this `sgml.Renderer` interface. Or at least for now. I thought we could initialize the SGML renderer by passing it the `templates` as you did here, but only once and at the main package level, based on the backend chosen by the user.

Reply:
I think it is potentially useful, as it allows for 3rd party applications to instantiate the HTML5 render and then just override certain templates, without having to subclass.

However, I guess you can also just copy the Templates and modify then call sgml.NewRenderer().

Reply:
I would rather wait until we have other SGML renderers to see if such an interface helps, to be honest.

Reply:
That's fine, I can remove this -- honestly this is mostly an artifact from earlier work on this code, and then I wound up creating the global Render() method and didn't need need the instances anymore.

My question still stands about whether you want me to wait for you to try fixing up the git history, or should I go ahead and fix the the items you've requested first?  I don't want to make the git history problem any worse if at all possible. :-)

Reply:
See my other comments -- I'm come back to my original position on this.  In fact, I am using this already in my xhtml5 work, and having the separate instances makes it possible to isolate state, permitting different formats to coexist in the same application.

So I vote *strongly* to keep this.  You'll almost certainly regret deciding not to later.

Reply:
Could you explain in which case you need to have separate instances (besides tests)? Why or when would you need to have different format coexisting in the same application?

Reply:
Imagine a hugo instance.  It can emit XHTML, or HTML5, or ePUB based on for example an Accept-Content: header, or some other tag.  This is all in the same application server.

In my case, I have a tool chain that renders different document formats depending on command line switches.  (My tool chain generates HTML, ePUB, and PDF.  Right now I rely on being able to spawn separate instances of the process, but since I'm building a composite document out of literally hundreds of individual .adoc files, I'd really really like to do this in one shot.)

So for me, this is very much *not* contrived.

Reply:
ok, so I'm not sure about the Hugo explanation, but I understand your use-case. Just to dig a little deeper on your toolchain: are you going to use the CLI or are you embedding the library elsewhere? because if you're using the CLI, then we'll probably need to support multiple backend values in the flags ;) (which is fine)

Reply:
I am using my own go program.

I'm actually checking the doc sources out from github into a memory file system using a go implementation of git, and then doing the work there.  There's a fair bit of processing to do (links get rewritten, etc.) and getting this all in one program makes the whole conversion run in 1-2s, instead of several minutes.

Reply:
> getting this all in one program makes the whole conversion run in 1-2s, instead of several minutes.

really?!? Wow, I'm impressed!